src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs

```diff
@@ -263,7 +263,29 @@ internal static string FindBestVersionMatch(VersionRange versionRange, IEnumerab
                 return version;
             }).Where(v => v != null);
 
-            var matchingVersion = bundleVersions.OrderByDescending(version => version.Version).FirstOrDefault();
+            var latestBundleVersion = bundleVersions.OrderByDescending(version => version.Version).FirstOrDefault();
+            var matchingVersion = latestBundleVersion;
+
+            if (string.Equals(_platformReleaseChannel, ScriptConstants.StandardPlatformChannelName, StringComparison.Ordinal) || string.Equals(_platformReleaseChannel, ScriptConstants.ExtendedPlatformChannelName, StringComparison.Ordinal))
+            {
+                if (bundleVersions.Count() > 1)
+                {
+                    matchingVersion = bundleVersions.OrderByDescending(version => version.Version).ElementAt(1); // if n is latest, should get version n-1
+                }
+                else
+                {
+                    // keep the latest version, log a warning
+                    _logger.LogInformation("Unable to retrieve the 2nd latest extension bundle version. The latest bundle version, {latestBundleVersion}, will be used instead.", latestBundleVersion);
```

Comment:
Log warning or keep as information? Not sure how often this would happen since this is a new feature.

Reply:
The wording here `2nd latest extension bundle version` can be confusing. We should try to keep the terms closer to the one used in the feature set itself.